FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -13,12 +13,42 @@ public final class RemoteFeedLoader: FeedLoader {
 		case invalidData
 	}
 
+	public struct Root: Decodable {
+		let items: [Item]
+	}
+
 	public init(url: URL, client: HTTPClient) {
 		self.url = url
 		self.client = client
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			
+			guard self != nil else { return }
+			
+			switch result {
+				case let .success((data, response)):
+					if response.statusCode == 200, let images = try? JSONDecoder().decode(Root.self, from: data) {
+						completion(.success(images.items.map({ $0.item })))
+					} else {
+						completion(.failure(Error.invalidData))
+					}
+				case .failure:
+					completion(.failure(Error.connectivity))
+			}
+			return
+		}
+	}
+}
+
+internal struct Item: Decodable {
```

Comment:
`internal` is a default access level, no need to declare it explicitly. 
Also this struct is a private implementation details thus it can be `private`

Please make `Item` private

```suggestion
private struct Item: Decodable {
```

As a suggestion `Item` declaration can be moved closer to `Root` so that both types describing api are close to each other. It makes them easier and faster to find.